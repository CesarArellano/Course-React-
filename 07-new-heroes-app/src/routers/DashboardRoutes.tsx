import { Route, Routes } from 'react-router-dom';

import { Navbar } from '../components/Navbar';
import { DCScreen } from '../components/dc/DCScreen';
import { MarvelScreen } from '../components/marvel/MarvelScreen';
import { SearchScreen } from '../components/search/SearchScreen';
import { HeroScreen } from '../components/hero/HeroScreen';

export const DashboardRoutes = () => {
  return (
    <>
      <Navbar />
      <Routes>
        <Route path="/marvel" element={ <MarvelScreen /> } />
        <Route path="/dc" element={ <DCScreen /> } />
        <Route path="/search" element={ <SearchScreen /> } />
        <Route path="/hero" element={ <HeroScreen /> } />
        <Route path="/" element={ <MarvelScreen /> } />
      </Routes>
    </>
  )
}
