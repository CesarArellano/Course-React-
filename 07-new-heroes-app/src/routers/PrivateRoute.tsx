import React, { useContext } from 'react'
import { Navigate } from 'react-router-dom';
import { AuthContext } from '../auth/authContext'

export const PrivateRoute = ({ children }: any) => {
  const { user } = useContext( AuthContext );
  
  return user.logged
    ? <> { children } </>
    : <Navigate to="/login "/>
}
