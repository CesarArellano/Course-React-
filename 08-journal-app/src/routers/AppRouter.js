import React, { useEffect, useState } from 'react'
import {
  BrowserRouter as Router,
  Switch,
  Route,
} from "react-router-dom";
import { JournalScreen } from '../components/journal/JournalScreen';
import { AuthRouter } from './AuthRouter';
import { firebase } from '../firebase/firebase-config';
import { useDispatch } from 'react-redux';
import { login } from '../actions/auth';

export const AppRouter = () => {

  const dispatch = useDispatch();
  const [checking, setChecking] = useState(true);
  const [ isLoggedIn, setIsLoggedIn ] = useState(false)
  
  useEffect(() => {
    firebase.auth().onAuthStateChanged( (user) => {
      if( user?.uid ) {
        dispatch(login(user.uid, user.displayName))
        setIsLoggedIn(true);
      } else {
        setIsLoggedIn(false);
      }
      setChecking(false);
    });
  }, [dispatch, setChecking]);

  if( checking ) {
    return (
      <div className="center-complete-box">
        <div className="utils__spinner"></div>
      </div>
    )
  }
  
  return (
    <Router>
      <Switch>
        <Route path="/auth" component={ AuthRouter } />
        <Route exact path="/" component={ JournalScreen } />
      </Switch>
    </Router>
  )
}
