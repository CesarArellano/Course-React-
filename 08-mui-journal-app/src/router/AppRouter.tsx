import { useSelector, useDispatch } from 'react-redux';
import { BrowserRouter, Navigate, Route, Routes } from 'react-router-dom'
import { AuthRoutes } from '../auth/routes/AuthRoutes';
import { JournalRoutes } from '../journal/routes/JournalRoutes';
import { CheckingAuth } from '../ui';
import { useEffect } from 'react';
import { onAuthStateChanged } from 'firebase/auth';
import { FirebaseAuth } from '../firebase/config';
import { login, logout } from '../store/auth';

export const AppRouter = () => {
  const { status } = useSelector<any, any>(state => state.auth);
  const dispatch = useDispatch<any>();
  useEffect(() => {
    onAuthStateChanged( FirebaseAuth, async (user) => {
      if(!user) return dispatch( logout(null) );
      dispatch( login(user) );
    })
  }, [])
  
  if( status === 'checking' ) {
    return <CheckingAuth />
  }

  return (
    <BrowserRouter>
      <Routes>
        {
          status === 'authenticated'
          ? <Route path='/*' element={ <JournalRoutes />} />            
          : <Route path='/auth/*' element={ <AuthRoutes />} />
        }
        <Route path='/*' element={ <Navigate to='auth/login' />} />  
      </Routes>
    </BrowserRouter>
  )
}
