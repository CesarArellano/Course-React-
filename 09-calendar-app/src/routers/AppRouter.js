import React from 'react'
import {
  BrowserRouter as Router,
  Switch,
  Route,
  Redirect
} from "react-router-dom";

import { LoginScreen } from '../components/auth/LoginScreen';
import { CalendarScreen } from '../components/calendar/CalendarScreen';

  // exact /login => LoginScreen
  // exact /  => Calendar
export const AppRouter = () => {
  return (
    <Router>
      <Switch>
        <Route exact path="/login" component={ LoginScreen }/>
        <Route exact path="/" component={ CalendarScreen }/>
        <Redirect to="/" />
      </Switch>
    </Router>
  )
}
