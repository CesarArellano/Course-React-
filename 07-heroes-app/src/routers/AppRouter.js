import React, { useContext } from 'react'
import {
  BrowserRouter as Router,
  Switch
} from "react-router-dom";
import { AuthContext } from '../auth/AuthContext';

import { LoginScreen } from '../components/login/LoginScreen';
import { DashboardRoutes } from './DashboardRoutes';
import { PublicRoute } from './PublicRoute';
import { PrivateRoute } from './PrivateRoute';

export const AppRouter = () => {
  const { user:{ logged } } = useContext(AuthContext);
  return (
    <Router>
      <div>
        <Switch>
          <PublicRoute 
            exact 
            path="/login" 
            isAuthenticated={ logged }
            component={ LoginScreen } 
          />
          <PrivateRoute 
            path="/" 
            isAuthenticated={ logged }
            component={ DashboardRoutes } 
          />
        </Switch>
      </div>
    </Router>
  );
}
