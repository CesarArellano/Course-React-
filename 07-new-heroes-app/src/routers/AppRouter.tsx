import { BrowserRouter, Route, Routes } from 'react-router-dom';

import { PrivateRoute } from './PrivateRoute';
import { LoginScreen } from '../components/login/LoginScreen';
import { PublicRoute } from './PublicRoute';
import { DashboardRoutes } from './DashboardRoutes';

export const AppRouter = () => {
  return (
    <BrowserRouter>
      <Routes>
        <Route path="/login" element={ 
          <PublicRoute>
            <LoginScreen />
          </PublicRoute>
          } 
        />
        <Route path="/*" element={ 
            <PrivateRoute>
              <DashboardRoutes />
            </PrivateRoute>
          }
        />
        {/* <Route path="/*" element={ <DashboardRoutes /> }/> */}
      </Routes>
    </BrowserRouter>
  );
}
