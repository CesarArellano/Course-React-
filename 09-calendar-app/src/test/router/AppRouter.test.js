import '@testing-library/jest-dom';

import React from 'react';
import configureStore from 'redux-mock-store';
import thunk from 'redux-thunk';
import { mount } from 'enzyme'
import { Provider } from 'react-redux';
import { AppRouter } from '../../routers/AppRouter';

const middlewares = [ thunk ];
const mockStore = configureStore( middlewares );

const initState = { 
  auth: {
    checking: true
  }
};

const store = mockStore( initState );

describe('Testing in <AppRouter />', () => {
  test('It should show waiting message', () => {
    const wrapper = mount(
      <Provider store={ store }>
        <AppRouter />
      </Provider>
    );
    
    expect( wrapper ).toMatchSnapshot();
    expect( wrapper.find('.utils__spinner').exists() ).toBe(true);
  })
});
