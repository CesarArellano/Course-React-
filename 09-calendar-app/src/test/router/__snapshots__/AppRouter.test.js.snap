// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing in <AppRouter /> It should show waiting message 1`] = `
<div
  className="center-complete-box"
>
  <div
    className="utils__spinner"
  />
</div>
`;
